<?xml version="1.0"?>
<launch>
    <arg name="tool_name" default="tool"/>
    <!-- Name of the tool we are going to use -->

    <!-- Parameters to pass to the ROS node -->
    <arg name="move_group" default="manipulator"/>
    <arg name="ee_link" default="tool_link_ee"/>
    <arg name="ros_rate" default="0.1"/>
    <arg name="udevname" default="/dev/ttyACM0"/>
    <arg name="portrate" default="9600"/>
    <arg name="use_conveyor" default="false"/>
    <arg name="use_gripper" default="false"/>
    <arg name="use_vision_sensor" default="false"/>
    <arg name="sim" default="false"/>

    <param name="/iiwa/toolName" type="string" value="$(arg tool_name)" />
    <!--Sunrise tool-->

    <!--  This loads the robot description  -->
    <include file="$(find iiwa_tool_moveit)/launch/moveit_planning_execution.launch">
        <arg name="sim" value="$(arg sim)"/>
    </include>

    <node pkg="iiwa_controller" type="iiwa_initial_pose_align.py" name="iiwa_initial_pose_align" output="screen" />
    <node pkg="iiwa_controller" type="iiwa_task_manager.py" name="iiwa_task_manager" output="screen">
        <param name="udevname" type="str" value="$(arg udevname)"/>
        <param name="portrate" type="int" value="$(arg portrate)"/>
        <param name="use_conveyor" type="bool" value="$(arg use_conveyor)"/>
        <param name="use_gripper" type="bool" value="$(arg use_gripper)"/>
        <param name="use_vision_sensor" type="bool" value="$(arg use_vision_sensor)"/>
    </node>
    <node ns="iiwa" name="iiwa_tcp_move_to_manipulate" pkg="iiwa_controller" type="iiwa_tcp_move_to_manipulate" respawn="false" output="screen">
        <param name="move_group" value="$(arg move_group)"/>
        <param name="ee_link" value="$(arg ee_link)"/>
        <param name="ros_rate" value="$(arg ros_rate)"/>
    </node>
    <node pkg="iiwa_controller" type="iiwa_commander.py" name="iiwa_commander" output="screen"/>
</launch>
